### PR TITLE
[BUGFIX CHORE tests] modernize unit/model-test (and some fixes that it uncovered)

### LINKED ISSUE #5679: Unable to peek created record from store by assigned id
Reproduction:
https://ember-twiddle.com/4520b9d2443bb9204d375b06d04ad530?openFiles=controllers.application.js%2C

Case:
I need to create totally new records, not yet persisted somewhere else, but whose ID is assigned by the client.

Working part:
```js
    let item2 = this.get('store').createRecord('user', { id: 43 });
    console.log('user:43', this.get('store').peekRecord('user', '43')); // --> item2 model
```
Broken part:
```js
    let item = this.get('store').createRecord('user');
    item.set('id', 42);
    console.log('user:42', this.get('store').peekRecord('user', '42')); //--> null
```

### PR DESCRIPTION
**Pull request by runspired**
resolves #5679

### CODE REVIEW

File: addon/-private/system/model/model.js [hunk 1]
```diff
@@ -456,7 +459,7 @@ const Model = EmberObject.extend(Evented, {
   */
   toJSON(options) {
     // container is for lazy transform lookups
-    let serializer = this.store.serializerFor('-default');
+    let serializer = this._internalModel.store.serializerFor('-default');
```
Comment (igorT) [inline addon/-private/system/model/model.js]: Why do we not lookup store directly anymore?

File: addon/-private/system/relationship-meta.js [hunk 2]
```diff
@@ -26,8 +26,7 @@ class RelationshipDefinition {
     this._type = '';
     this.__inverseKey = '';
     this.__inverseIsAsync = null;
-    this.modelClass = meta.parentType;
-    this.store = null;
```
Comment (igorT) [inline addon/-private/system/relationship-meta.js]: Is this never set?

Comment (runspired) [reply addon/-private/system/relationship-meta.js]: `store` was never set.

File: addon/-private/system/model/model.js (hunk 1 above)
Comment (runspired) [reply addon/-private/system/model/model.js]: We only accessed `store` two places from `Model` ourselves. Moving these two places to `_internalModel` limits our exposure. We now no longer care whether `store` is properly assigned or not. We give it to the model but end users may override as it is not necessary to stay there.

**Pull request merged**